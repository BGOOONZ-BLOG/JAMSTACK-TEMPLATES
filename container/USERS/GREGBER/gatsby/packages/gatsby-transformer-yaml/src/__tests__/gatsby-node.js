const Promise = require(`bluebird`)
const os = require(`os`)
const { onCreateNode } = require(`../gatsby-node`)

const createNodeId = jest.fn().mockReturnValue(`uuid-from-gatsby`)
const createContentDigest = jest.fn().mockReturnValue(`contentDigest`)

const loadNodeContent = node => Promise.resolve(node.content)

let createNode
let createParentChildLink
let actions
let node
let createNodeSpec

beforeEach(() => {
  createNode = jest.fn()
  createParentChildLink = jest.fn()
  actions = { createNode, createParentChildLink }

  node = {
    id: `whatever`,
    parent: null,
    children: [],
    internal: {
      contentDigest: `whatever`,
      mediaType: `text/yaml`,
    },
  }

  createNodeSpec = {
    node,
    loadNodeContent,
    actions,
    createNodeId,
    createContentDigest,
  }
})

const expectCreatedNodesMatchingSnapshot = count => {
  expect(createNode.mock.calls).toMatchSnapshot()
  expect(createParentChildLink.mock.calls).toMatchSnapshot()
  expect(createNode).toHaveBeenCalledTimes(count)
  expect(createParentChildLink).toHaveBeenCalledTimes(count)
}

const expectCreatedNodeTypeName = nodeTypeName => {
  expect(createNode).toBeCalledWith(
    expect.objectContaining({
      internal: expect.objectContaining({
        type: nodeTypeName,
      }),
    })
  )
}

describe(`Processing YAML nodes with internal type 'File'`, () => {
  beforeEach(() => {
    node.internal.type = `File`
    node.dir = `${os.tmpdir()}/top/foo bar/`
    node.name = `My File`
  })

  describe(`from a single YAML dictionary,`, () => {
    beforeEach(() => {
      node.content = `
            id: some
            blue: true
            funny: yup
      `
    })

    it(`when no target type name is specified, the type name is based on the directory name`, async () => {
      await onCreateNode(createNodeSpec)

      expectCreatedNodesMatchingSnapshot(1)
      expectCreatedNodeTypeName(`FooBarYaml`)
    })

    it(`with the specified target type name`, async () => {
      await onCreateNode(createNodeSpec, {
        typeName: `fixed`,
      })

      expectCreatedNodesMatchingSnapshot(1)
      expectCreatedNodeTypeName(`fixed`)
    })

    it(`with the target type name explicitly retrieved from the YAML content`, async () => {
      // noinspection JSUnusedLocalSymbols (unused fields left as documentation)
      await onCreateNode(createNodeSpec, {
        typeName: ({ node, object, isArray }) => object.funny,
      })

      expectCreatedNodesMatchingSnapshot(1)
      expectCreatedNodeTypeName(`yup`)
    })
  })

  describe(`from a YAML array of dictionaries,`, () => {
    beforeEach(() => {
      node.content = `
            - blue: true
              funny: yup
            - blue: false
              funny: nope
      `
    })

    it(`when no target type name is specified, all nodes have a type name based on the file name`, async () => {
      await onCreateNode(createNodeSpec)

      expectCreatedNodesMatchingSnapshot(2)
      expectCreatedNodeTypeName(`MyFileYaml`)
      expectCreatedNodeTypeName(`MyFileYaml`)
    })

    it(`with the specified target type name`, async () => {
      await onCreateNode(createNodeSpec, {
        typeName: `fixed`,
      })

      expectCreatedNodesMatchingSnapshot(2)
      expectCreatedNodeTypeName(`fixed`)
      expectCreatedNodeTypeName(`fixed`)
    })

    it(`with the target type name explicitly retrieved from the YAML content`, async () => {
      // noinspection JSUnusedLocalSymbols (unused fields left as documentation)
      await onCreateNode(createNodeSpec, {
        typeName: ({ node, object, isArray }) => object.funny,
      })

      expectCreatedNodesMatchingSnapshot(2)
      expectCreatedNodeTypeName(`yup`)
      expectCreatedNodeTypeName(`nope`)
    })
  })
})

describe(`Processing YAML nodes with internal type other than 'File'`, () => {
  beforeEach(() => {
    node.internal.type = `Other`
  })

  describe(`from a single YAML dictionary,`, () => {
    beforeEach(() => {
      node.content = `
            id: some
            blue: true
            funny: yup
      `
    })

    it(`when no target type name is specified, the type name is based on the original node's internal type`, async () => {
      await onCreateNode(createNodeSpec)

      expectCreatedNodesMatchingSnapshot(1)
      expectCreatedNodeTypeName(`OtherYaml`)
    })

    it(`with the specified target type name`, async () => {
      await onCreateNode(createNodeSpec, {
        typeName: `fixed`,
      })

      expectCreatedNodesMatchingSnapshot(1)
      expectCreatedNodeTypeName(`fixed`)
    })

    it(`with the target type name explicitly retrieved from the YAML content`, async () => {
      // noinspection JSUnusedLocalSymbols (unused fields left as documentation)
      await onCreateNode(createNodeSpec, {
        typeName: ({ node, object, isArray }) => object.funny,
      })

      expectCreatedNodesMatchingSnapshot(1)
      expectCreatedNodeTypeName(`yup`)
    })
  })

  describe(`from a YAML array of dictionaries,`, () => {
    beforeEach(() => {
      node.content = `
            - blue: true
              funny: yup
            - blue: false
              funny: nope
      `
    })

    it(`when no target type name is specified, all nodes have a type name based on the original node's internal type`, async () => {
      await onCreateNode(createNodeSpec)

      expectCreatedNodesMatchingSnapshot(2)
      expectCreatedNodeTypeName(`OtherYaml`)
      expectCreatedNodeTypeName(`OtherYaml`)
    })

    it(`with the specified target type name`, async () => {
      await onCreateNode(createNodeSpec, {
        typeName: `fixed`,
      })

      expectCreatedNodesMatchingSnapshot(2)
      expectCreatedNodeTypeName(`fixed`)
      expectCreatedNodeTypeName(`fixed`)
    })

    it(`with the target type name explicitly retrieved from the YAML content`, async () => {
      // noinspection JSUnusedLocalSymbols (unused fields left as documentation)
      await onCreateNode(createNodeSpec, {
        typeName: ({ node, object, isArray }) => object.funny,
      })

      expectCreatedNodesMatchingSnapshot(2)
      expectCreatedNodeTypeName(`yup`)
      expectCreatedNodeTypeName(`nope`)
    })
  })
})
