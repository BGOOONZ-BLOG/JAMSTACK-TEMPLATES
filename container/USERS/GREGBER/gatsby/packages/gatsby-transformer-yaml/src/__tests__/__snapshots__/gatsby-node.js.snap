// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Processing YAML nodes with internal type 'File' from a YAML array of dictionaries, when no target type name is specified, all nodes have a type name based on the file name 1`] = `
Array [
  Array [
    Object {
      "blue": true,
      "children": Array [],
      "funny": "yup",
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "MyFileYaml",
      },
      "parent": "whatever",
    },
  ],
  Array [
    Object {
      "blue": false,
      "children": Array [],
      "funny": "nope",
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "MyFileYaml",
      },
      "parent": "whatever",
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type 'File' from a YAML array of dictionaries, when no target type name is specified, all nodes have a type name based on the file name 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "blue": true,
        "children": Array [],
        "funny": "yup",
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "MyFileYaml",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            - blue: true
              funny: yup
            - blue: false
              funny: nope
      ",
        "dir": "<TEMP_DIR>/top/foo bar/",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "File",
        },
        "name": "My File",
        "parent": null,
      },
    },
  ],
  Array [
    Object {
      "child": Object {
        "blue": false,
        "children": Array [],
        "funny": "nope",
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "MyFileYaml",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            - blue: true
              funny: yup
            - blue: false
              funny: nope
      ",
        "dir": "<TEMP_DIR>/top/foo bar/",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "File",
        },
        "name": "My File",
        "parent": null,
      },
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type 'File' from a YAML array of dictionaries, with the specified target type name 1`] = `
Array [
  Array [
    Object {
      "blue": true,
      "children": Array [],
      "funny": "yup",
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "fixed",
      },
      "parent": "whatever",
    },
  ],
  Array [
    Object {
      "blue": false,
      "children": Array [],
      "funny": "nope",
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "fixed",
      },
      "parent": "whatever",
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type 'File' from a YAML array of dictionaries, with the specified target type name 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "blue": true,
        "children": Array [],
        "funny": "yup",
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "fixed",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            - blue: true
              funny: yup
            - blue: false
              funny: nope
      ",
        "dir": "<TEMP_DIR>/top/foo bar/",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "File",
        },
        "name": "My File",
        "parent": null,
      },
    },
  ],
  Array [
    Object {
      "child": Object {
        "blue": false,
        "children": Array [],
        "funny": "nope",
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "fixed",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            - blue: true
              funny: yup
            - blue: false
              funny: nope
      ",
        "dir": "<TEMP_DIR>/top/foo bar/",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "File",
        },
        "name": "My File",
        "parent": null,
      },
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type 'File' from a YAML array of dictionaries, with the target type name explicitly retrieved from the YAML content 1`] = `
Array [
  Array [
    Object {
      "blue": true,
      "children": Array [],
      "funny": "yup",
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "yup",
      },
      "parent": "whatever",
    },
  ],
  Array [
    Object {
      "blue": false,
      "children": Array [],
      "funny": "nope",
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "nope",
      },
      "parent": "whatever",
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type 'File' from a YAML array of dictionaries, with the target type name explicitly retrieved from the YAML content 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "blue": true,
        "children": Array [],
        "funny": "yup",
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "yup",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            - blue: true
              funny: yup
            - blue: false
              funny: nope
      ",
        "dir": "<TEMP_DIR>/top/foo bar/",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "File",
        },
        "name": "My File",
        "parent": null,
      },
    },
  ],
  Array [
    Object {
      "child": Object {
        "blue": false,
        "children": Array [],
        "funny": "nope",
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "nope",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            - blue: true
              funny: yup
            - blue: false
              funny: nope
      ",
        "dir": "<TEMP_DIR>/top/foo bar/",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "File",
        },
        "name": "My File",
        "parent": null,
      },
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type 'File' from a single YAML dictionary, when no target type name is specified, the type name is based on the directory name 1`] = `
Array [
  Array [
    Object {
      "blue": true,
      "children": Array [],
      "funny": "yup",
      "id": "some",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "FooBarYaml",
      },
      "parent": "whatever",
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type 'File' from a single YAML dictionary, when no target type name is specified, the type name is based on the directory name 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "blue": true,
        "children": Array [],
        "funny": "yup",
        "id": "some",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "FooBarYaml",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            id: some
            blue: true
            funny: yup
      ",
        "dir": "<TEMP_DIR>/top/foo bar/",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "File",
        },
        "name": "My File",
        "parent": null,
      },
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type 'File' from a single YAML dictionary, with the specified target type name 1`] = `
Array [
  Array [
    Object {
      "blue": true,
      "children": Array [],
      "funny": "yup",
      "id": "some",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "fixed",
      },
      "parent": "whatever",
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type 'File' from a single YAML dictionary, with the specified target type name 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "blue": true,
        "children": Array [],
        "funny": "yup",
        "id": "some",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "fixed",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            id: some
            blue: true
            funny: yup
      ",
        "dir": "<TEMP_DIR>/top/foo bar/",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "File",
        },
        "name": "My File",
        "parent": null,
      },
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type 'File' from a single YAML dictionary, with the target type name explicitly retrieved from the YAML content 1`] = `
Array [
  Array [
    Object {
      "blue": true,
      "children": Array [],
      "funny": "yup",
      "id": "some",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "yup",
      },
      "parent": "whatever",
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type 'File' from a single YAML dictionary, with the target type name explicitly retrieved from the YAML content 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "blue": true,
        "children": Array [],
        "funny": "yup",
        "id": "some",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "yup",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            id: some
            blue: true
            funny: yup
      ",
        "dir": "<TEMP_DIR>/top/foo bar/",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "File",
        },
        "name": "My File",
        "parent": null,
      },
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type other than 'File' from a YAML array of dictionaries, when no target type name is specified, all nodes have a type name based on the original node's internal type 1`] = `
Array [
  Array [
    Object {
      "blue": true,
      "children": Array [],
      "funny": "yup",
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "OtherYaml",
      },
      "parent": "whatever",
    },
  ],
  Array [
    Object {
      "blue": false,
      "children": Array [],
      "funny": "nope",
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "OtherYaml",
      },
      "parent": "whatever",
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type other than 'File' from a YAML array of dictionaries, when no target type name is specified, all nodes have a type name based on the original node's internal type 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "blue": true,
        "children": Array [],
        "funny": "yup",
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "OtherYaml",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            - blue: true
              funny: yup
            - blue: false
              funny: nope
      ",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "Other",
        },
        "parent": null,
      },
    },
  ],
  Array [
    Object {
      "child": Object {
        "blue": false,
        "children": Array [],
        "funny": "nope",
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "OtherYaml",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            - blue: true
              funny: yup
            - blue: false
              funny: nope
      ",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "Other",
        },
        "parent": null,
      },
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type other than 'File' from a YAML array of dictionaries, with the specified target type name 1`] = `
Array [
  Array [
    Object {
      "blue": true,
      "children": Array [],
      "funny": "yup",
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "fixed",
      },
      "parent": "whatever",
    },
  ],
  Array [
    Object {
      "blue": false,
      "children": Array [],
      "funny": "nope",
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "fixed",
      },
      "parent": "whatever",
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type other than 'File' from a YAML array of dictionaries, with the specified target type name 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "blue": true,
        "children": Array [],
        "funny": "yup",
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "fixed",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            - blue: true
              funny: yup
            - blue: false
              funny: nope
      ",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "Other",
        },
        "parent": null,
      },
    },
  ],
  Array [
    Object {
      "child": Object {
        "blue": false,
        "children": Array [],
        "funny": "nope",
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "fixed",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            - blue: true
              funny: yup
            - blue: false
              funny: nope
      ",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "Other",
        },
        "parent": null,
      },
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type other than 'File' from a YAML array of dictionaries, with the target type name explicitly retrieved from the YAML content 1`] = `
Array [
  Array [
    Object {
      "blue": true,
      "children": Array [],
      "funny": "yup",
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "yup",
      },
      "parent": "whatever",
    },
  ],
  Array [
    Object {
      "blue": false,
      "children": Array [],
      "funny": "nope",
      "id": "uuid-from-gatsby",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "nope",
      },
      "parent": "whatever",
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type other than 'File' from a YAML array of dictionaries, with the target type name explicitly retrieved from the YAML content 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "blue": true,
        "children": Array [],
        "funny": "yup",
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "yup",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            - blue: true
              funny: yup
            - blue: false
              funny: nope
      ",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "Other",
        },
        "parent": null,
      },
    },
  ],
  Array [
    Object {
      "child": Object {
        "blue": false,
        "children": Array [],
        "funny": "nope",
        "id": "uuid-from-gatsby",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "nope",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            - blue: true
              funny: yup
            - blue: false
              funny: nope
      ",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "Other",
        },
        "parent": null,
      },
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type other than 'File' from a single YAML dictionary, when no target type name is specified, the type name is based on the original node's internal type 1`] = `
Array [
  Array [
    Object {
      "blue": true,
      "children": Array [],
      "funny": "yup",
      "id": "some",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "OtherYaml",
      },
      "parent": "whatever",
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type other than 'File' from a single YAML dictionary, when no target type name is specified, the type name is based on the original node's internal type 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "blue": true,
        "children": Array [],
        "funny": "yup",
        "id": "some",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "OtherYaml",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            id: some
            blue: true
            funny: yup
      ",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "Other",
        },
        "parent": null,
      },
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type other than 'File' from a single YAML dictionary, with the specified target type name 1`] = `
Array [
  Array [
    Object {
      "blue": true,
      "children": Array [],
      "funny": "yup",
      "id": "some",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "fixed",
      },
      "parent": "whatever",
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type other than 'File' from a single YAML dictionary, with the specified target type name 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "blue": true,
        "children": Array [],
        "funny": "yup",
        "id": "some",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "fixed",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            id: some
            blue: true
            funny: yup
      ",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "Other",
        },
        "parent": null,
      },
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type other than 'File' from a single YAML dictionary, with the target type name explicitly retrieved from the YAML content 1`] = `
Array [
  Array [
    Object {
      "blue": true,
      "children": Array [],
      "funny": "yup",
      "id": "some",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "yup",
      },
      "parent": "whatever",
    },
  ],
]
`;

exports[`Processing YAML nodes with internal type other than 'File' from a single YAML dictionary, with the target type name explicitly retrieved from the YAML content 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "blue": true,
        "children": Array [],
        "funny": "yup",
        "id": "some",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "yup",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "content": "
            id: some
            blue: true
            funny: yup
      ",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "text/yaml",
          "type": "Other",
        },
        "parent": null,
      },
    },
  ],
]
`;
